tests/client.test.ts

```diff
@@ -9,3 +12,102 @@ describe("Client Creation", () => {
     expect(client.chain).toBe(simulator);
   });
 });
+
+describe("Client Overrides", () => {
+  it("should default to client account if no account is provided", async () => {
+    const account = createAccount(generatePrivateKey());
+    const client = createClient({
+      chain: simulator,
+      account,
+    });
+
+    // Mock the client.request method
+    vi.spyOn(client, "request").mockResolvedValue(undefined);
+
+    const contractAddress = "0x1234567890123456789012345678901234567890";
+    await client.readContract({
+      address: contractAddress as Address,
+      functionName: "testFunction",
+      args: ["arg1", "arg2"],
+    });
+
+    expect(client.request).toHaveBeenCalledWith({
+      method: "eth_call",
+      params: [
+        {
+          to: contractAddress,
+          from: account.address,
+          data: expect.any(String),
+        },
+        "latest",
+      ],
+    });
+  });
+
+  it("should override client account if account is provided", async () => {
+    const account = createAccount(generatePrivateKey());
+    const client = createClient({
+      chain: simulator,
+      account,
+    });
+
+    const overrideAccount = createAccount(generatePrivateKey());
+
+    // Mock the client.request method
+    vi.spyOn(client, "request").mockResolvedValue(undefined);
+
+    const contractAddress = "0x1234567890123456789012345678901234567890";
+    await client.readContract({
+      account: overrideAccount,
+      address: contractAddress as Address,
+      functionName: "testFunction",
+      args: ["arg1", "arg2"],
+    });
+
+    expect(client.request).toHaveBeenCalledWith({
+      method: "eth_call",
+      params: [
+        {
+          to: contractAddress,
+          from: overrideAccount.address,
+          data: expect.any(String),
+        },
+        "latest",
+      ],
+    });
+  });
+});
+
+describe("Request Method", () => {
```

Comment:
@AgustinRamiroDiaz it seems that the expects are missing here

Reply:
@cristiam86 these are examples of what I wanted to ask. In these tests, I only want to test the typing of the `request` method. How can I test that? If I put any garbage in it, the test still passes